File: Hospital_Management_Microservice/Hospital_Management_NET/Hospital_Management/Models/Doctor.cs
using System;
using System.Collections.Generic;

namespace Hospital_Management.Models;

public partial class Doctor
{
    public long Id { get; set; }

    public double Amount { get; set; }

    public string? Degree { get; set; }

    public byte[]? Doctorimage { get; set; }

    public string? Email { get; set; }

    public string? Name { get; set; }

    public string? Password { get; set; }

    public string? Phone { get; set; }

    public string? Role { get; set; }

    public long? SpecializationId { get; set; }

    public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();

    public virtual Specialization? Specialization { get; set; }
}

File: Hospital_Management_Microservice/hospital management_springboot/src/main/java/com/app/Entity/Doctor.java
package com.app.Entity;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.Lob;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Entity
@Getter
@Setter
@ToString(exclude = { "doctorimage", "specialization", "appointments" })
public class Doctor {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	private String name;
	private String phone;
	private String degree;
	private double amount;
	private String email;
	private String password;

	@Enumerated(EnumType.STRING)
	private Role role; // DOCTOR

	@Lob
	@Column(columnDefinition = "LONGBLOB") // To store image as BLOB
	private byte[] doctorimage;

	@ManyToOne
	@JoinColumn(name = "specialization_id")
	private Specialization specialization;

	@OneToMany(mappedBy = "doctor", cascade = CascadeType.ALL)
	private List<Appointment> appointments;

}
